### ISSUE TITLE
local path in requirements.yaml with file:// is relative to working directory, not chart

### ISSUE THREAD

**Issue by chancez** [kubernetes/helm#2103]
I have a repo containing a few levels of subcharts like so:
```
.
├── Chart.yaml
├── Jenkinsfile
├── README.md
├── charts
│   ├── app-deploy
│   │   ├── Chart.yaml
│   │   ├── charts
│   │   ├── requirements.lock
│   │   ├── requirements.yaml
│   │   ├── templates
│   │   ├── values.yaml
│   │   └── vendored_charts
│   └── app-migrate
│       ├── Chart.yaml
│       ├── charts
│       ├── templates
│       └── values.yaml
```

My requirements.yaml is this:
```
dependencies:
- name: nginx-ingress
  version: 0.3.1
  repository: file://./vendored_charts/nginx-ingress
  condition: nginx-ingress.enabled
```

I typically run my commands in the root of the repo like this: `helm upgrade ./charts/app-deploy`. 

However, if I run `helm dep update ./charts/app-deploy` I get this error:
```
Error: directory /Users/chance/projects/charts/website/vendored_charts/nginx-ingress not found
```

If i cd into charts:
```
Error: directory /Users/chance/projects/charts/website/charts/vendored_charts/nginx-ingress not found
```

It then works if I cd into `charts/app-deploy` and run `helm dep update`.

https://github.com/kubernetes/helm/pull/1923/files#diff-ed5c3d716ac6aec0a4fd3d6628e281d2R66 isn't taking into the account when the path is relative, it should be `path.Join()`ing the `depPath` to the path of the chart whose dependencies are being updated.

**Comment by qwangrepos** [comment]
Thank you for reporting this. I'll submit a patch to fix it.

**Comment by technosophos** [comment]
Fixed in #2194
